Harvesting (issue 1205)

dposner-dayspring opened the pull request:
create generic message for reset password request.

dposner-dayspring committed:
Harvesting (issue 1205)
create generic message for reset password request.

jwong-dayspring commented on Controller/ForgotResetController.php:
```diff
@@ -52,6 +52,8 @@ public function __construct(
      */
     public function forgotPasswordAction(Request $request)
     {
+        $generic_msg = 'Your request has been sent.';
```
Per Dayspring standard practice, properties should be `camelCase`.

ssitu-dayspring replied:
@jwong-dayspring Do you think we should also tack on the text: "Please check your email for further instructions.", or if it's more ideal to keep the generic msg as is?

jwong-dayspring replied:
@ssitu-dayspring It's probably good to be helpful in the text. Something like "If an account was found, an email has been sent. Please check your email for further instructions."

jwong-dayspring commented on Controller/ForgotResetController.php:
```diff
@@ -93,19 +95,19 @@ public function forgotPasswordAction(Request $request)
 
                     $request->getSession()->getFlashBag()->add(
                         "success",
```
In addition to using the same message, we should use the same type of flash bag to avoid hinting that an email address exists. Consider using the "success" type for all cases. (This probably means code could be further refactored to avoid duplicate code.)

jwong-dayspring commented on Controller/ForgotResetController.php:
```diff
@@ -93,19 +95,19 @@ public function forgotPasswordAction(Request $request)
-                        "Check your email for instructions on how to reset your password."
+                        $generic_msg //"Check your email for instructions on how to reset your password."
                     );
                     return $this->redirect($this->generateUrl('_login'));
```
We should also be consistent about redirecting the user to avoid hinting at success.

jwong-dayspring reviewed (changes requested):
See comments, and also failing tests.

dposner-dayspring committed:
Harvesting (issue 1205)
create generic message for reset password request.
set flash status as success and redirect to login page.  refactor code.

dposner-dayspring committed:
Harvesting (issue 1205)
create generic message for reset password request.
fix test

dposner-dayspring committed:
Harvesting (issue 1205)
create generic message for reset password request.
fix test

dposner-dayspring committed:
Harvesting (issue 1205)
create generic message for reset password request.
fix test

dposner-dayspring committed:
Harvesting (issue 1205)
create generic message for reset password request.
fix test

dposner-dayspring committed:
Harvesting (issue 1205)
update generic message for reset password request.

ssitu-dayspring commented:
@jwong-dayspring I got an autowire error when I updated composer on BSN.

> Cannot autowire service "Dayspring\LoginBundle\Controller\ForgotResetController": argument "$mailer" of method "__construct()" references interface "Symfony\Component\Mailer\MailerInterface" but no such service exists. Did you create a class that implements this interface?

Do you know how stable is the 5.1.4 release, or if it's compatible with Symfony 4.4? I think **master** is currently tagged with 5.1.4.

Maybe we should keep this separate from **master** and branch off from 4.1.0 instead?

jwong-dayspring commented:
> @jwong-dayspring I got an autowire error when I updated composer on BSN.
> 
> > Cannot autowire service "Dayspring\LoginBundle\Controller\ForgotResetController": argument "$mailer" of method "__construct()" references interface "Symfony\Component\Mailer\MailerInterface" but no such service exists. Did you create a class that implements this interface?
> 
> Do you know how stable is the 5.1.4 release, or if it's compatible with Symfony 4.4? I think **master** is currently tagged with 5.1.4.
> 
> Maybe we should keep this separate from **master** and branch off from 4.1.0 instead?

Hmm, I think you're right that this should be branched from 4.1.0. Maybe a "4.x" branch. The current master branch has changes to make LoginBundle compatible with Symfony 5.x, and that included removing Swiftmailer and replacing it with the MailerInterface.

ssitu-dayspring reviewed (dismissed):
rebased branch to 4.1.0 release

jwong-dayspring committed:
rebuild om before test

jwong-dayspring committed:
add missing build.properties

jwong-dayspring committed:
add docker dev environment

jwong-dayspring committed:
Merge branch '4.x' into feature/pen_test_remediation

jwong-dayspring committed:
Merge branch 'master' into feature/pen_test_remediation
